src/tezos-client/index.ts

```diff
@@ -25,6 +25,31 @@ export const prepareTransaction = (
   };
 };
 
+export const prepareTransactionForSubmission = (
+  payment: BasePayment
+): WalletParamsWithKind => {
+  return {
+    kind: OpKind.TRANSACTION,
+    to: payment.recipient,
+    amount: payment.amount.toNumber(),
+    mutez: true,
+    /* For current simplicity, submit non-delegator payments without the estimates */
+    fee:
+      payment.type === EPaymentType.Delegator
```

Comment:
Is there reason why we do this for `EPaymentType.Delegator` txs only? 

I would remove this restriction.

Reply:
Currently, we are estimating fees only for delegator payments. Fee income and bond reward payments do not have estimates attached. 

Making estimates for bond rewards and fee income is a wider refactor that I would scope out of this PR. I'm not sure it has a purpose in any case – i.e. there is no need to deduct fees from baker's payment to himself. 

I can perhaps make my comment clearer?